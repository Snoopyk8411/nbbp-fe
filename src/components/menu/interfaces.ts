import { Appearance } from 'tools/types/api-catalog-items-types';
import { SubmenuOpenActionType, SubmenuPosition } from './constants';

export interface IMenuProps {
  items: IMenuItemModel[];
  submenuPosition: SubmenuPosition;
}

export interface IMenuLevelProps {
  items: IMenuItemModel[];
  parentItem?: IMenuItemModel;
  submenuOpenActionType: SubmenuOpenActionType;
  submenuPosition: SubmenuPosition;
  containerElement: HTMLElement | null;
  onClose?: () => void;
}

export type MenuItemIdType = string | number;

export interface IMenuItemModel {
  id: MenuItemIdType;
  name: string;
  order: number;
  url: string;
  appearance: Appearance | String;
  children?: IMenuItemModel[];
  svgName: string | undefined;
}
