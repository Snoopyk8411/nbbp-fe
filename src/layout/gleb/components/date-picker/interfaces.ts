export interface IDateData {
  year: number;
  month: number;
  date: number;
}
