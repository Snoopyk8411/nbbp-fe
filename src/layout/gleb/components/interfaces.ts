import { AnyAction, Action, Store, Middleware } from 'redux';

export interface IMockStore<S = any, A extends Action = AnyAction> extends Store<S, A> {
  getActions(): any[];
  clearActions(): void;
}

export type MockStoreEnhanced<S = {}, DispatchExts = {}> = IMockStore<S> & { dispatch: DispatchExts };

export type MockStoreCreator<S = {}, DispatchExts = {}> = (
  state?: S | MockGetState<S>,
) => MockStoreEnhanced<S, DispatchExts>;

export type MockGetState<S = {}> = (actions: AnyAction[]) => S;

/**
 * Create Mock Store returns a function that will create a mock store from a state
 * with the same set of set of middleware applied.
 *
 * @param middlewares The list of middleware to be applied.
 * @template S The type of state to be held by the store.
 * @template DispatchExts The additional Dispatch signatures for the middlewares applied.
 */
declare function createMockStore<S, DispatchExts = {}>(middlewares?: Middleware[]): MockStoreCreator<S, DispatchExts>;

export default createMockStore;

export interface IMockData {
  mediaPage: {
    media: {
      copyright: string;
      date: string;
      explanation: string;
      hdurl: string;
      media_type: string;
      service_version: string;
      title: string;
      url: string;
    };
    error: Error | null;
    selectedDay: any;
    isLoading: boolean;
  };
}
