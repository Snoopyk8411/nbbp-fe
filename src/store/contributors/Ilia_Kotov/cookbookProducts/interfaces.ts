export interface IProduct {
  id: number | string;
  name: string;
}

export type IProductsData = IProduct[];
