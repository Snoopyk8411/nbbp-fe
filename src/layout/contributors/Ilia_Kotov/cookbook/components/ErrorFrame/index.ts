import dynamic from 'next/dynamic';

export const ErrorFrame = dynamic(() => import('./ErrorFrame'), { ssr: false });
