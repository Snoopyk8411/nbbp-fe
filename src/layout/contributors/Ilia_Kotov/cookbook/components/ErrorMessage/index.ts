import dynamic from 'next/dynamic';

export const ErrorMessage = dynamic(() => import('./ErrorMessage'), { ssr: false });
