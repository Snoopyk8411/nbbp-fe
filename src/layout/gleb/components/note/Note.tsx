type Props = {
  note: string;
};

const Note: React.FC<Props> = ({ note }) => {
  return <li>{note}</li>;
};

export default Note;
