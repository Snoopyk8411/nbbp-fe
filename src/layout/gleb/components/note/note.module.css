.notes {
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  height:100%
}

.title {
  margin-left: 1.5rem
}

.input {
  background-color: rgb(229, 231, 235);
  border-color: rgb(209, 213, 219);
  border-width: 2px;
  outline: none;
  padding: 0.5rem 1rem;
  color: #1f2937;
}
.input:focus {
  background-color: darkgrey;
}
