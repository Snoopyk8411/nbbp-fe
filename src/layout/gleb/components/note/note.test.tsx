import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { Note } from 'layout/gleb/components/note/Note';
import { IMockStore } from 'layout/gleb/components/interfaces';

let mockStore: IMockStore;
const mockStoreConf = configureStore();
const mockData = {};

describe('Note Component', () => {
  let props: { note: string };
  beforeEach(() => {
    mockStore = mockStoreConf(mockData);
    props = {
      note: 'write a note',
    };
  });

  it('makes Note snapshot', () => {
    const component = renderer.create(
      <Provider store={mockStore}>
        <Note {...props} />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
