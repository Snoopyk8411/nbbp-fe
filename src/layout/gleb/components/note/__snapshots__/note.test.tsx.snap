// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note Component Note snapshot 1`] = `
<li>
  write a note
</li>
`;
